src/EventStore.Common/Configuration/OptionsDumper.cs

```diff
@@ -26,19 +26,39 @@ public class OptionsDumper {
 			var modified = info.Where(x => x.Value.source != typeof(Default));
 
 			var nameColumnWidth = info.Keys.Select(x => x.Length).Max() + 7;
-
+			
 			return PrintOptions(nameof(modified), modified) + Environment.NewLine +
 			       PrintOptions(nameof(@default), @default);
 
-			string PrintOptions(string name,
-				IEnumerable<KeyValuePair<string, (Type source, string value)>> options) =>
-				options.OrderBy(pair => _sortOrder.IndexOf(pair.Key)).Aggregate(
-					new StringBuilder().Append($"{name.ToUpper()} OPTIONS:").AppendLine().AppendLine(),
-					(builder, pair) => builder
-						.Append($"     {pair.Key}:".PadRight(nameColumnWidth, ' '))
-						.Append(pair.Value.value).Append(' ').Append(FormatSourceName(pair.Value.source))
-						.AppendLine()).ToString();
-
+			string PrintOptions(string name, IEnumerable<KeyValuePair<string, (Type source, string value)>> configOptions) {
+				var dumpOptions = new StringBuilder();
+				dumpOptions.Append($"{name.ToUpper()} OPTIONS:").AppendLine().AppendLine();
+				foreach (var groupOptions  in _sortGroupBy) {
+					bool oneTimeCheckForGroupOptions = true;
+					
+					foreach (var option in groupOptions) {
+						var result = (from configOption in configOptions
+							where string.Equals(
+								NameTranslators.RemoveWhiteSpacesToCombine(configOption.Key, string.Empty), option.Name,
+								StringComparison.CurrentCultureIgnoreCase)
+							select configOption).FirstOrDefault();
```

Comment:
it feels like what we could do with here is a dictionary lookup rather than searching through the `configOptions` each time. the funny thing is in `Dump`, `info` already _is_ a dictionary. So perhaps this will turn out nicely if we pass `info` in here as `configOptions` and also pass a `Func<Type, bool>` to act as a filter.

Out in `Dump` the two filters can be defined something like like 

```
			bool Default(Type x) => x == typeof(Default);
			bool Modified(Type x) => x != typeof(Default);
```